core/src/main/scala/zquery/ZQuery.scala

```diff
@@ -199,6 +201,24 @@ sealed trait ZQuery[-R, +E, +A] { self =>
       a     <- runCache(cache)
     } yield (cache, a)
 
+  /**
+   * Summarizes a query by computing some value before and after execution,
+   * and then combining the values to produce a summary, together with the
+   * result of execution.
+   */
+  final def summarized[R1 <: R, E1 >: E, B, C](f: (B, B) => C)(summary: ZIO[R1, E1, B]): ZQuery[R1, E1, (C, A)] =
```

Comment:
Shall we switch the parameters here too?

Reply:
Yes. I was slightly inclined to wait until RC18 and make conforming changes to that so we track the public ZIO API for identical methods but happy to change now as well.

Reply:
Ah ok, either way's fine 👍